### PR TITLE
small updates for cuda, soft->prob, int assert

### PR DESCRIPTION
**Pull request by paroma**
__`.cuda()` to `.to(device)` fixes #135__ 
- set default to be cpu
- removed 1-2 if statements
- we have to check `!= cpu` instead of `==cuda` since it can be `cuda` or `cuda:X`

__`soft/hard labels` to `probs/preds` fixes #66__ 
- changes made throughout notebooks, READMEs and code

__fixing assert for int labels fixes #81__
- added assert statement to check type of `Y`
- checking Torch.Tensor() is messy because it makes everything a float when you convert

### CODE REVIEW

**Comment by bhancock8** [comment]
This is great! Looks like it's not passing the formatting tests right now. Did you run `make fix` before submitting?

File: metal/classifier.py [hunk 1]
```diff
@@ -200,10 +200,9 @@ def _train_model(
         epoch_size = len(train_loader.dataset)
 
         # Move model to GPU
-        if self.config["use_cuda"]:
-            if self.config["verbose"]:
-                print("Using GPU...")
-            self.cuda()
+        if self.config["verbose"] and self.config["device"] != "cpu":
+            print("Using GPU...")
+        self.to(self.config["device"])
```
Comment (jdunnmon) [inline metal/classifier.py]: I could be missing something, but I think this needs an indent to be inside the `if` statement?

File: metal/classifier.py [hunk 2]
```diff
@@ -419,7 +418,7 @@ def _create_data_loader(self, data, **kwargs):
 
     def _set_seed(self, seed):
         self.seed = seed
-        if torch.cuda.is_available():
+        if self.config["device"] != "cpu":
             # TODO: confirm this works for gpus without knowing gpu_id
```
Comment (jdunnmon) [inline metal/classifier.py]: Can we get rid of this `TODO` now?
